src/Event/AbstractStorageEventListener.php

```diff
@@ -137,16 +139,9 @@ public function buildFilename($table, $entity) {
  * @return string
  */
 	public function buildPath($table, $entity) {
-		$path = '';
-		if ($this->_config['tableFolder']) {
-			$path .= $table->table() . DS;
-		}
-		if ($this->_config['randomPath'] === true) {
-			$path .= StorageUtils::randomPath($entity[$table->primaryKey()]);
-		}
-		if ($this->_config['uuidFolder'] === true) {
-			$path .= $this->stripDashes($entity[$table->primaryKey()]) . DS;
-		}
+		$pathBuilder = $this->createPathBuilder($entity['adapter']);
```

Comment:
Mmh. I think you lost me there … But why having two locations where paths are being built? On the one hand there is a path builder that is used everywhere (LocalListener, FileStorageTable, AbstractListener, …). Also in the comments of Storageutils:random() you state " * @deprecated Use the randomPath() method from the BasePathBuilder instead.” Hence the switch to path builder …

Bernhard


> On 12.08.2015, at 10:01, Florian Krämer <notifications@github.com> wrote:
> 
> In src/Event/AbstractStorageEventListener.php <https://github.com/burzum/cakephp-file-storage/pull/77#discussion_r36834857>:
> 
> > @@ -137,16 +139,9 @@ public function buildFilename($table, $entity) {
> >   * @return string
> >   */
> >  	public function buildPath($table, $entity) {
> > -		$path = '';
> > -		if ($this->_config['tableFolder']) {
> > -			$path .= $table->table() . DS;
> > -		}
> > -		if ($this->_config['randomPath'] === true) {
> > -			$path .= StorageUtils::randomPath($entity[$table->primaryKey()]);
> > -		}
> > -		if ($this->_config['uuidFolder'] === true) {
> > -			$path .= $this->stripDashes($entity[$table->primaryKey()]) . DS;
> > -		}
> > +		$pathBuilder = $this->createPathBuilder($entity['adapter']);
> This is not going to work because the "adapter" (which should have been named config in the past) is a configuration name that can be anything. It was never thought to be used as a convention. Also the idea is in fact that the path builder doesn't have to match the adapter name to allow people to customize the path building. I won't accept this part of the changes.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/burzum/cakephp-file-storage/pull/77/files#r36834857>.
>